### ISSUE TITLE
re2 failed to install on Node 16

### ISSUE THREAD

**Issue by RaresAil** [uhop/node-re2#100]
With `yarn add re2` i get this issue on Node 16

```
error /Users/rares/.config/yarn/global/node_modules/re2: Command failed.
Exit code: 1
Command: install-from-cache --artifact build/Release/re2.node --host-var RE2_DOWNLOAD_MIRROR || npm run rebuild
Arguments: 
Directory: /Users/rares/.config/yarn/global/node_modules/re2
Output:
Trying https://github.com/uhop/node-re2/releases/download/1.15.9/darwin-x64-93.br ...
Trying https://github.com/uhop/node-re2/releases/download/1.15.9/darwin-x64-93.gz ...
Building locally ...

> re2@1.15.9 rebuild
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.0.0 | darwin | x64
gyp info find Python using Python version 2.7.16 found at "/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python"
(node:6735) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn /System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
gyp info spawn args [
gyp info spawn args   '/Users/rares/.config/yarn/global/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/rares/.config/yarn/global/node_modules/re2/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/rares/.config/yarn/global/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/rares/Library/Caches/node-gyp/16.0.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/rares/Library/Caches/node-gyp/16.0.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/rares/.config/yarn/global/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/rares/Library/Caches/node-gyp/16.0.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/rares/.config/yarn/global/node_modules/re2',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/re2/lib/addon.o
In file included from ../lib/addon.cc:1:
In file included from ../lib/./wrapped_re2.h:4:
In file included from ../../nan/nan.h:56:
In file included from /Users/rares/Library/Caches/node-gyp/16.0.0/include/node/node.h:63:
In file included from /Users/rares/Library/Caches/node-gyp/16.0.0/include/node/v8.h:30:
/Users/rares/Library/Caches/node-gyp/16.0.0/include/node/v8-internal.h:452:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                ~~~~~^~~~~~~~~~~
                                     remove_cv
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:697:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
1 error generated.
make: *** [Release/obj.target/re2/lib/addon.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/rares/.config/yarn/global/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 20.3.0
gyp ERR! command "/usr/local/Cellar/node/16.0.0/bin/node" "/Users/rares/.config/yarn/global/node_modules/re2/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/rares/.config/yarn/global/node_modules/re2
gyp ERR! node -v v16.0.0
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok 
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "1".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

> re2@1.15.9 rebuild
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.0.0 | darwin | x64
gyp info find Python using Python version 2.7.16 found at "/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python"
(node:6784) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn /System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
gyp info spawn args [
gyp info spawn args   '/Users/rares/.config/yarn/global/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/rares/.config/yarn/global/node_modules/re2/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/rares/.config/yarn/global/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/rares/Library/Caches/node-gyp/16.0.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/rares/Library/Caches/node-gyp/16.0.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/rares/.config/yarn/global/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/rares/Library/Caches/node-gyp/16.0.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/rares/.config/yarn/global/node_modules/re2',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/re2/lib/addon.o
In file included from ../lib/addon.cc:1:
In file included from ../lib/./wrapped_re2.h:4:
In file included from ../../nan/nan.h:56:
In file included from /Users/rares/Library/Caches/node-gyp/16.0.0/include/node/node.h:63:
In file included from /Users/rares/Library/Caches/node-gyp/16.0.0/include/node/v8.h:30:
/Users/rares/Library/Caches/node-gyp/16.0.0/include/node/v8-internal.h:452:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                ~~~~~^~~~~~~~~~~
                                     remove_cv
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:697:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
1 error generated.
make: *** [Release/obj.target/re2/lib/addon.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/rares/.config/yarn/global/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 20.3.0
gyp ERR! command "/usr/local/Cellar/node/16.0.0/bin/node" "/Users/rares/.config/yarn/global/node_modules/re2/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/rares/.config/yarn/global/node_modules/re2
gyp ERR! node -v v16.0.0
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok
```

**Comment by uhop** [comment]
It is already fixed by #98, yet the fix is unpublished.

**Comment by uhop** [comment]
Publishing the fix has failed (https://github.com/uhop/node-re2/actions/runs/785999037) due to two things:

* There is no official docker image for Node 16, which is used by the build system on Linux.
* New `node-gyp` fails on Linux in mysterious ways.

I plan to address these issues before publishing the new version, otherwise I'll ship those problems to end users.

**Comment by radicand** [comment]
> Publishing the fix has failed (https://github.com/uhop/node-re2/actions/runs/785999037) due to two things:
> 
>     * There is no official docker image for Node 16, which is used by the build system on Linux.

FWIW node16 docker builds just went up on Dockerhub

**Comment by keepitterron** [comment]
ping :)

**Comment by uhop** [comment]
Released in 1.16.0.

Sorry for taking too long. Node 16 required a new `node-gyp`, which had a major release, and finally switched to python3. It broke all custom actions for Linux platform. It didn't help that Github Actions are acting up now periodically failing with 502 on getting files.

It helped a little to retire a build for Node 10, which EOLed a week ago. The next one is Node 15, which will EOL in a month.
